### DISCUSSION TITLE
Extracting Frontmatter data from a different MD file

### DISCUSSION THREAD

**Discussion by gruff101** [SilentVoid13/Templater#1113] [Help] [answered]
Hi,

I followed the example below to extract values out of variables which so far are in the same file.

https://silentvoid13.github.io/Templater/internal-functions/internal-modules/frontmatter-module.html

I would like to extract variables out of a different file but I am struggling to do this. 

How do you use tp.frontmatter on a different file?

**Comment by Zachatoo** [comment, marked as answer]
Here's one way to do that:

```js
<%*
const file = tp.file.find_tfile("Note with frontmatter");
const fileCache = await app.metadataCache.getFileCache(file);
const frontmatter = fileCache?.frontmatter;
_%>

...

<% frontmatter.status %>
```

More examples [here](https://zachyoung.dev/snippets/templater#retrieve-frontmatter-from-another-note)

**Reply by gruff101** [reply]
Thanks @Zachatoo code worked perfectly.
